staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -417,6 +422,13 @@ loop:
 			*resourceVersion = newResourceVersion
 			r.setLastSyncResourceVersion(newResourceVersion)
 			eventCount++
+
+		case <-time.After(60 * time.Second):
+			utilruntime.HandleError(fmt.Errorf("%s: unable to receive heartbeat watch event, maybe caused by dead connection", r.name))
+			if r.onHeartbeatTimeout != nil {
+				r.onHeartbeatTimeout()
+			}
+			break loop
```

Comment:
does this kick us out to a re-list, or a re-watch?

Reply:
re-list